### ISSUE TITLE
[BUG] Archive/Merged.txt files grow to ludicrous size

### ISSUE THREAD

**Issue by Atreidae** [Tzahi12345/YoutubeDL-Material#777]
**Describe the bug**
Frequently YTDL-Material (actually, I think its YTDL itself that does it) will insert the same records into the Archive_video.txt and Merged_Video.txt files. This causes _ERR string too-long_ errors when YTDL_Matrieal tries to process the file info

``` Error: Cannot create a string longer than 0x1fffffe8 characters
    at Buffer.utf8Slice (<anonymous>)
    at Object.slice (node:buffer:594:37)
    at Buffer.toString (node:buffer:812:14)
    at FSReqCallback.readFileAfterClose [as oncomplete] (node:internal/fs/read_file_context:63:23) {
  code: 'ERR_STRING_TOO_LONG'
```

**To Reproduce**
Steps to reproduce the behaviour:
Allow YTDL to scrape a large number of channels... this might be related to YTDL restarting when there are too many downloads waiting in the database (which causes an endless loop of YTDL restarting)

**Expected behaviour**
Only one entry per video in the db files

**Screenshots**
![image](https://user-images.githubusercontent.com/8736291/199900778-affc75e3-11f9-40e5-a579-ebe56baf9321.png)

**Environment**
- YoutubeDL-Material version 4.3 Stable
- Docker tag: Latest

**Workaround**

I've thrown together a quick and nasty PowerShell script that will remove any duplicate lines from each file, 
It's not fast or, efficient, but it gets there slowly. 
Use at your own risk and backup your Archives folder first (they compress really nicely)

```
# Dirty ytdl cleanup script

$library = "\\172.16.1.93\dockerbulk\youtubedl-material\subscriptions" #folder to search

#####

cd "$library\archives"

Write-host "Finding Files"
#Find all the files we care about
$filestoProcess = (get-ChildItem -include archive.txt,archive_video.txt,merged_video.txt -Recurse)

#process each item
Write-host "processing files"
$i=0
Foreach ($file in $filestoProcess) {
  $i++
  Write-host "File $i of $($filestoprocess.count)"


  [string]$filename = $file.fullname
  Write-host "$filename"
  
  $Content = (Get-Content -LiteralPath $file | Sort-Object | Get-Unique)
  Remove-item $File
  $content |Out-File -FilePath $file -Append -NoClobber -Encoding default
}
```
![image](https://user-images.githubusercontent.com/8736291/199900358-66db1e33-3139-4d69-ae32-7ce2da5706eb.png)

**Comment by Atreidae** [comment]
Okay, maybe PowerShell wasnt the best way to sweep through these files..
![image](https://user-images.githubusercontent.com/8736291/199933195-fbb37dc0-2ae5-4c08-b724-2ac399c29950.png)

**Comment by Atreidae** [comment]
The filesize bug appears to be related to downloading a lot of files from the same channel. it should be noted that #758 points out a race condition where multiple download threads start. I'd suggest this may also be inserting the same entries over and over again in the DB txt files.

This is just from one night.
![image](https://user-images.githubusercontent.com/8736291/200197997-c4eb92f5-17a3-4cd0-b3d1-5c6809eb92fd.png)

**Comment by Atreidae** [comment]
Found a much faster text processor than my dodgy PowerShell implementation, which brings a 1GB file down to 2kb in about a minute.
See option 2 here https://www.raymond.cc/blog/remove-duplicate-lines-with-textcrawler-scratchpad-or-notepad/#:~:text=8%20Ways%20To%20Remove%20Duplicate%20Lines%20in%20Text,Remove%20Duplicate%20Lines%20Online%20...%208%208.%20TextMechanic

**Comment by Tzahi12345** [comment]
This will be fixed with #809! We are moving away from using files for archives and instead storing that information in the DB

**Comment by GlassedSilver** [comment]
Are entries cleaned up before they are imported from these files?

Secondary question, what about completely derped lines, e.g. lines that are actually man hundreds of lines stored without line delimiter resulting in a super long id string on one line?

I inteded to work on fixing archives I know. but once I figured out how difficult it is to sanity check these IDs against duplicates when taking into account that they are completely broken at times... I felt like I needed to plan that a lot more thoroughly and that was when I basically took a break from the project both in terms of using it and in terms of developing the fix scripts. I simpley realized I wanted to work on a solution and grow with the task, but it simply exceeded my skill, not my will to learn however. (but on the other hand certainly my time to... yikes)

Or are the IDs for archives pulled from somewhere else for the db? Are we file-scanning the directories and looking for downloads actually on-disk? That could be a cool way to do it, probably not as reliable thogh if the user didn't use IDs in their filenaming scheme. :thinking:

**Comment by Tzahi12345** [comment]
I took a super long break too haha so I don't blame you. I wrote up a quick discussion post so we can move the convo over there: https://github.com/Tzahi12345/YoutubeDL-Material/discussions/898

I included some of the code I used for the importing, and a quick description of how it works. I do like your "scan through the DB and generate the archive from there" idea but I don't wanna force it on people.
